engine/engine.go

```diff
@@ -357,11 +358,20 @@ func (p *Engine) Stop(ctx context.Context, d notification.Dest) error {
 		return errors.New("stop only supported on user contact methods")
 	}
 	var err error
+	contactMethodType := contactmethod.TypeFromDestType(d.Type)
 
 	permission.SudoContext(ctx, func(ctx context.Context) {
-		err = p.cfg.ContactMethodStore.DisableByValue(ctx, contactmethod.TypeFromDestType(d.Type), d.Value)
+		err = p.cfg.ContactMethodStore.DisableByValue(ctx, contactMethodType, d.Value)
 	})
 
+	// NOTE TCPA compliance: maintain a record of consent/dissent
+	if err == nil {
+		userID := permission.UserID(ctx)
+		time := time.Now().String()
+		logStr := fmt.Sprintf("user %s stopped %s notifications at %s", userID, contactMethodType, time)
```

Comment:
`log.Logf` takes its first argument as a format string. But use `WithFields` for the ID and keep the message fixed in this case.

- We'll need to return the CM id from the query above, UserID can probably be omitted, as the CM id already links to the user ID
- The context is likely not associated with the user, since we're using `SudoContext` above (meaning `permission.UserID` will most likely return an empty string)
- The log text should just be something fixed like "Contact method STOP received." or something of that nature. 
- This should probably be in the store method instead, in case it's ever called from elsewhere. Anything that triggers Enable/Disable should log. That way we can also avoid adding/changing the API when returning the contact method ID